### ISSUE TITLE
masked softmax?

### ISSUE THREAD

**Issue by hockman1** [keras-team/keras#14316]
Hi I would like to implement masked softmax function. For example, if y = [1,2,3,4], I would like to perform softmax over the first 2 indices, then masked_softmax(y) should give [0.268, 0.731, 0, 0], but I am not too sure how to implement this as a activation function when training a deep neural network? Help pls thanx!

**Comment by hockman1** [comment]
I tried this:

def custom_softmax(t):
    
    a = np.zeros((2,2))
    a = np.reshape(a,(2,2)) # Sideway concatenation
    b = K.softmax(t[:,:2])
    
    return K.concatenate((b,a),axis=1)

However, it doesnt seem to backpropagate when training a network

**Comment by yash-clear** [comment]
@hockman1 
`
inp = np.asarray([1., 2.,3., 4.])
layer = tf.keras.layers.Softmax()
layer(inp).numpy()
mask = np.asarray([True,True,False,False], dtype=bool)
layer(inp, mask).numpy()
`
output:
array([0.26894143, 0.7310586 , 0.        , 0.        ], dtype=float32)

**Comment by yash-clear** [comment]
I want to make my first contribution tell if it is right!

**Comment by hockman1** [comment]
Looks good thanks!

**Comment by sushreebarsa** [comment]
@hockman1 Could you please let us know if you still face the issue ? If this is resolved for you then please move this issue to closed status .Thanks!

**Comment by sushreebarsa** [comment]
Closing this issue due to lack of recent activity & assuming this issue as resolved .Please feel free to reopen if you still have concern.Thanks!
